server/commands/validators/awsAccountValidator.js

```diff
@@ -26,10 +28,19 @@ function validate(account) {
     validateAccountNumber(account.AccountNumber);
 
     if (account.IsMaster && masterAccount !== undefined && account.AccountNumber !== masterAccount.AccountNumber)
-      throw new Error(`The account '${masterAccount.AccountName}' is already set as the master account.`)
+      throw new Error(`The account '${masterAccount.AccountName}' is already set as the master account.`);
+
+  }).then(co.wrap(function* () {
+    if (account.RoleArn !== null) {
+      try {
+        let roleIsValid = yield childAWSclient.assumeRole(account.RoleArn);
+      } catch (error) {
+        throw new Error(`${account.AccountName} is not permitted to assume the required role`);
```

Comment:
May be worth logging the original error in the catch block so the stack trace isn't lost?